### PR TITLE
Card spending control

### PR DESCRIPTION
**Pull request by lera**
Implement new endpoints  [Cards](https://docs.solarisgroup.com/api-reference/digital-banking/cards/#operation/createCardControlSpendingLimit) 

- Create card spending limit control
- Index card spending limit controls
- Delete card spending limit control

2. update validateCardLimits()
3. update card ui with new limits
4. everything is tested via postman

Screen Shot 2023-02-16 at 17 31 17

### CODE REVIEW

File: src/helpers/reservations.ts [hunk 1]
```diff
@@ -281,167 +283,130 @@ const computeCardUsage = (person: MockPerson) => {
...
+    currentCardUsage.ATM.daily.amount > dailyATMLimitAmount ||
+    currentCardUsage.ATM.daily.count > dailyATMLimitCount ||
+    currentCardUsage.ATM.monthly.amount > monthlyATMLimitAmount ||
+    currentCardUsage.ATM.monthly.count > monthlyATMLimitCount ||
+    currentCardUsage.PURCHASE.daily.amount > dailyPurchaseLimitAmount ||
+    currentCardUsage.PURCHASE.daily.count > dailyPurchaseLimitCount ||
+    currentCardUsage.PURCHASE.monthly.amount > monthlyPurchaseLimitAmount ||
+    currentCardUsage.PURCHASE.monthly.count > monthlyPurchaseLimitCount
+  ) {
+    // triggerCardDeclinedWebhook
```
Comment (lera) [inline src/helpers/reservations.ts:408]: @mbayraktr you will need to trigger new webhook from here with the reason `SPENDING_LIMIT`

Comment (mbayraktr) [reply src/helpers/reservations.ts:408]: thank you @lera for the info 🙏

File: src/helpers/cards.ts [hunk 2]
```diff
@@ -763,4 +766,83 @@ export const updateCardSettings = async (
   return settings;
 };
 
+export const createCardSpendingLimit = async (
+  req,
+  res
+): Promise<CardSpendingLimitControl> => {
+  const { scope_id: cardId, limit, idempotency_key: idempotencyKey } = req.body;
```
Comment (mbayraktr) [inline src/helpers/cards.ts]: Don't we need `scope` param in body as defined in doc ?
Screen Shot 2023-02-17 at 17 25 41

File: src/helpers/cards.ts [hunk 3]
```diff
@@ -763,4 +766,83 @@ export const updateCardSettings = async (
...
+          status: HttpStatusCodes.NOT_FOUND,
+          code: "model_not_found",
+          title: "Model Not Found",
+          detail: `Couldn't find 'Solaris::CardAccount' for id '${cardId}'.`,
+        },
+      ],
+    });
+  }
+
+  const cardControl = cardData.controls?.find(
```
Comment (mbayraktr) [inline src/helpers/cards.ts:797]: I am not sure if the `idempotency_key` check is enough to understand if the limit is already created.

Comment (lera) [reply src/helpers/cards.ts:797]: I've tried on sandbox and solaris continue pushing same limit into the card control. They just use the lowest during validation

File: src/helpers/cards.ts (hunk 2 above)
Comment (lera) [reply src/helpers/cards.ts]: we will use only CARD scope and there is no need to add more
we always add only the necessary logic 
I can return [] if the scope is not CARD

**Review by mbayraktr — APPROVED** [review]

**Pull request merged**